contracts/interfaces/IAutomateProxyFactory.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+interface IAutomateProxyFactory {
+    /**
+     * @notice Emitted when an OpsProxy is deployed.
+     *
+     * @param deployer Address which initiated the deployment
+     * @param owner The address which the proxy is for.
+     * @param proxy Address of deployed proxy.
+     */
+    event DeployProxy(
+        address indexed deployer,
+        address indexed owner,
+        address indexed proxy
+    );
+
+    /**
+     * @notice Emitted when OpsProxy implementation to be deployed is changed.
+     *
+     * @param oldImplementation Previous OpsProxy implementation.
+     * @param newImplementation Current OpsProxy implementation.
+     */
+    event SetImplementation(
+        address indexed oldImplementation,
+        address indexed newImplementation
+    );
+
+    /**
+     * @notice Emitted when OpsProxy implementation is added or removed from whitelist.
+     *
+     * @param implementation OpsProxy implementation.
+     * @param whitelisted Added or removed from whitelist.
+     */
+    event UpdateWhitelistedImplementation(
+        address indexed implementation,
+        bool indexed whitelisted
+    );
+
+    /**
+     * @notice Deploys OpsProxy for the msg.sender.
+     *
+     * @return proxy Address of deployed proxy.
+     */
+    function deploy() external returns (address payable proxy);
+
+    /**
+     * @notice Deploys OpsProxy for another address.
+     *
+     * @param owner Address to deploy the proxy for.
+     *
+     * @return proxy Address of deployed proxy.
+     */
+    function deployFor(address owner) external returns (address payable proxy);
+
+    /**
+     * @notice Sets the OpsProxy implementation that will be deployed by OpsProxyFactory.
```

Comment:
Consider changing these all together for consistency?

- `OpsProxyFactory` -> `AutomateProxyFactory`
- `OpsProxy` -> `AutomateProxy`

Not necessary though.

Reply:
Agree, would be nice to get this renamed as well while we are at it